lib/modules/ipfs/index.js

```diff
@@ -66,20 +58,31 @@ class IPFS {
         }
         self.logger.trace("IPFS available");
         return cb({name: "IPFS ", status: 'on'});
-      }
+      });
     });
   }
 
-  addProviderToEmbarkJS() {
-    const self = this;
-    // TODO: make this a shouldAdd condition
-    if (this.storageConfig === {}) {
-      return;
+  _getNodeUrl() {
+    if (this.storageConfig.upload.provider === 'ipfs') {
+      let {protocol, host, port} = this.storageConfig.upload;
+      return (protocol || 'http') + '://' + host + ':' + port + '/api/v0/version';
```

Comment:
use `utils.buildUrl` or `utils.buildUrlFromConfig` here to properly handle when port === false.